Source/WebCore/dom/Node.h

```diff
@@ -349,6 +349,8 @@ class Node : public EventTarget, public CanMakeCheckedPtr<Node> {
     // Returns the enclosing event parent Element (or self) that, when clicked, would trigger a navigation.
     WEBCORE_EXPORT Element* enclosingLinkEventParentOrSelf();
 
+    WEBCORE_EXPORT void expandAllDetailsAncestors();
```

Comment:
This doesn't feel like it should be a `Node` member function to me. Maybe it can be a free function in `HTMLDetailsElement.cpp` or in `FindController.cpp` or perhaps a static member function of one of the classes. It doesn't need any `private` or `protected` access to `Node` and doesn’t seem like a fundamental `Node` operation.